flaml/ml.py

```diff
@@ -319,8 +319,11 @@ def evaluate_model_CV(
         shuffle = False
     elif isinstance(kf, TimeSeriesSplit):
         kf = kf.split(X_train_split, y_train_split)
-    else:
+    elif hasattr(kf, "split"):
         kf = kf.split(X_train_split)
+    else:
+        n = len(kf)

```

Comment:
any constraint on the iterable object `split_type` in automl.py, such that kf is already in `train_index, test_index` ndarray?